### ISSUE TITLE
LLVM_ERROR with -Zmir-opt-level=3: match.rs

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust#75507]
Code from ./src/test/codegen/match.rs
```rust
#![crate_type = "lib"]

pub enum E {
    A,
    B,
}

#[no_mangle]
pub fn exhaustive_match(e: E) -> u8 {
    match e {
        E::A => 0,
        E::B => 1,
    }
}

```

There was an LLVM ERROR with
`rustc ./src/test/codegen/match.rs  -Zmir-opt-level=3`

```
Function return type does not match operand type of return inst!
  ret i1 %7
 i8in function exhaustive_match
LLVM ERROR: Broken function found, compilation aborted!
```
### Meta

repo @ 5e3f1b148db5bfa27fee52464ae1f5d34c49d77b

<details><summary>Backtrace</summary>
<p>

```
Function return type does not match operand type of return inst!
  ret i1 %7
 i8in function exhaustive_match
LLVM ERROR: Broken function found, compilation aborted!
```

</p>
</details>

**Comment by matthiaskrgr** [comment]
Probably related to https://github.com/rust-lang/rust/pull/75382

**Comment by JulianKnodt** [comment]
Should be fixed by #75537
